onnxruntime/test/onnx/TestCase.cc

```diff
@@ -589,3 +566,57 @@ OnnxTestCase::OnnxTestCase(const std::string& test_case_name, _In_ std::unique_p
     return true;
   });
 }
+
+void LoadTests(const std::vector<std::basic_string<PATH_CHAR_TYPE>>& input_paths,
+               const std::vector<std::basic_string<PATH_CHAR_TYPE>>& whitelisted_test_cases,
+               double default_per_sample_tolerance, double default_relative_per_sample_tolerance,
+               const std::unordered_set<std::basic_string<ORTCHAR_T>>& disabled_tests,
+               const std::function<void(std::unique_ptr<ITestCase>)>& process_function) {
+  std::vector<std::basic_string<PATH_CHAR_TYPE>> paths(input_paths);
+  while (!paths.empty()) {
+    std::basic_string<PATH_CHAR_TYPE> node_data_root_path = paths.back();
+    paths.pop_back();
+    std::basic_string<PATH_CHAR_TYPE> my_dir_name = GetLastComponent(node_data_root_path);
+    LoopDir(node_data_root_path, [&](const PATH_CHAR_TYPE* filename, OrtFileType f_type) -> bool {
+      if (filename[0] == '.') return true;
+      if (f_type == OrtFileType::TYPE_DIR) {
+        std::basic_string<PATH_CHAR_TYPE> p = ConcatPathComponent<PATH_CHAR_TYPE>(node_data_root_path, filename);
+        paths.push_back(p);
+        return true;
+      }
+
+      std::basic_string<PATH_CHAR_TYPE> filename_str = filename;
+#if !defined(ORT_MINIMAL_BUILD)
+      if (!HasExtensionOf(filename_str, ORT_TSTR("onnx")))
+        return true;
+#else
+      if( !HasExtensionOf(filename_str, ORT_TSTR("ort")) )
+         return true;
+#endif
+
+      std::basic_string<PATH_CHAR_TYPE> test_case_name = my_dir_name;
+      if (test_case_name.compare(0, 5, ORT_TSTR("test_")) == 0) test_case_name = test_case_name.substr(5);
+
+      if (!whitelisted_test_cases.empty() && std::find(whitelisted_test_cases.begin(), whitelisted_test_cases.end(),
+                                                       test_case_name) == whitelisted_test_cases.end()) {
+        return true;
+      }
+      if (disabled_tests.find(test_case_name) != disabled_tests.end()) return true;
+
+      std::basic_string<PATH_CHAR_TYPE> p = ConcatPathComponent<PATH_CHAR_TYPE>(node_data_root_path, filename_str);
+
+      std::unique_ptr<TestModelInfo> model_info;
+#if !defined(ORT_MINIMAL_BUILD)
+      model_info = TestModelInfo::LoadOnnxModel(p.c_str());
```

Comment:
How is the error while loading model handled?

Reply:
I think an exception is thrown. This code has not been changed. I just saved it out of a file that got deleted.

---
In reply to: [489063767](https://github.com/microsoft/onnxruntime/pull/5169#discussion_r489063767) [](ancestors = 489063767)

Reply:
No I meant once exception is thrown by ORT how does the test code handle it

Reply:
This depends on the caller. There are two cases: 1) onnx_test_runner main() has a try/catch 2)  model_ tests just Googletest framework handle it.

---
In reply to: [489155922](https://github.com/microsoft/onnxruntime/pull/5169#discussion_r489155922) [](ancestors = 489155922)